c/secp256k1_blake160_sighash_all.c

```diff
@@ -110,39 +91,50 @@ int main(int argc, char* argv[]) {
       return ERROR_WITNESS_TOO_LONG;
     }
 
-    if (!(witness_table = ns(Witness_as_root(witness)))) {
-      return ERROR_ENCODING;
-    }
-    args = ns(Witness_data(witness_table));
-    if (ns(Bytes_vec_len(args)) < 1) {
-      return ERROR_WRONG_NUMBER_OF_ARGUMENTS;
-    }
+    witness_pos.ptr = (const uint8_t*)witness;
+    witness_pos.size = len;
 
     /* Load signature */
-    len = TEMP_SIZE;
-    ret = extract_bytes(ns(Bytes_vec_at(args, 0)), temp, &len);
-    if (ret != CKB_SUCCESS) {
+    arg_res = mol_cut(&witness_pos, MOL_Witness(0));
+    if (arg_res.code != 0) {
+      if (arg_res.attr < 1) {
+        return ERROR_WRONG_NUMBER_OF_ARGUMENTS;
+      } else {
+        return ERROR_ENCODING;
+      }
+    }
+    witness_len = arg_res.attr;
+
+    bytes_res = mol_cut_bytes(&arg_res.pos);
+    if (bytes_res.code != 0) {
+      return ERROR_ENCODING;
+    } else if (bytes_res.pos.size < 65) {
```

Comment:
Define 65 as a constant `SIGNATURE_SIZE`

Reply:
I just inherited this from previous commits~ Maybe next PR?